### ISSUE TITLE
Ship TypeScript types for the packaged frontend config

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1046]
## Version

`@utoo/lint@0.2.5`

## Context

The package exports a first-party frontend preset at `@utoo/lint/configs/frontend`, but the export points directly to `frontend.json` without a TypeScript declaration.

Using the preset from a typed `utlint.config.ts` therefore fails type checking:

```ts
import { defineConfig } from "@utoo/lint/config";
import frontend from "@utoo/lint/configs/frontend";

export default defineConfig({
  rules: {
    ...frontend.rules,
  },
});
```

Run:

```bash
tsc --noEmit \
  --module esnext \
  --moduleResolution bundler \
  --resolveJsonModule \
  --allowSyntheticDefaultImports \
  utlint.config.ts
```

## Actual behavior

TypeScript reports:

```text
Type '{ "no-debugger": string; ... }' is not assignable to type
'Record<string, RuleConfig>'.

Property '"no-debugger"' is incompatible with index signature.
Type 'string' is not assignable to type 'RuleConfig'.
```

JSON imports widen the preset severities to `string`, so the first-party preset is not assignable to `ConfigObject`.

The current workaround is a manual assertion:

```ts
import type { ConfigObject } from "@utoo/lint";

const frontendRules =
  frontend.rules as NonNullable<ConfigObject["rules"]>;
```

## Expected behavior

The packaged preset should be directly usable from `utlint.config.ts` without a type assertion, for example:

```ts
export default defineConfig({
  rules: {
    ...frontend.rules,
  },
});
```

One possible implementation is to ship a typed JS/TS preset entry or a `frontend.d.ts`, then expose it through a conditional `types` entry for `./configs/frontend`.

The TypeScript configuration docs should also show how to extend/import the first-party frontend preset without copying it into a JSON config.

## Adoption context

Found while migrating the Utoo documentation site in [utooland/site#21](https://github.com/utooland/site/pull/21).
